### PR TITLE
Fix applicationinsights.json config is not loaded in manual rp attach scenario

### LINKED ISSUE #3283: The configuration file present in the same directory as the agent is not being picked-up in 3.4.16
### Expected behavior
Default configuration file `applicationinsights.json` present in the same directory as the agent should be automatically picked-up by the agent when the application starts.

### Actual behavior
The default application insights configuration is being applied and the configuration file is being ignored.

### To Reproduce
* Provide a configuration file that sits next to the agent where you change the selfDiagnostics level to DEBUG.
* Do NOT specify the path to the configuration file using the environment variable `APPLICATIONINSIGHTS_CONFIGURATION_FILE`

```
"selfDiagnostics": {
  "destination": "console",
  "level": "DEBUG"
}
```

### System information
Please provide the following information:
 - SDK Version: 3.4.16
 - OS type and version: Debian GNU/Linux 11 (bullseye)
 - Application Server type and version (if applicable): OpenJDK Runtime Environment 18.9 (build 11.0.16+8)
 - Using spring-boot? Yes

### Logs
```
Picked up JAVA_TOOL_OPTIONS: -Xms512m -Xmx1024m -XX:MaxMetaspaceSize=300m -javaagent:/applicationinsights-agent-3.4.16.jar
OpenJDK 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
2023-09-11 09:05:27.791Z INFO c.m.applicationinsights.agent - Application Insights Java Agent 3.4.16 started successfully (PID 1, JVM running for 3.386 s)
2023-09-11 09:05:27.796Z INFO c.m.applicationinsights.agent - Java version: 11.0.16, vendor: Oracle Corporation, home: /usr/local/openjdk-11
```

### PR DESCRIPTION
**Pull request by heyams**
Fix #3283

### CODE REVIEW

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java [hunk 1]
```diff
@@ -478,7 +479,9 @@ private static Configuration loadConfigurationFile(Path agentJarPath) {
       return getConfiguration(runtimeAttachedConfigurationContent, JsonOrigin.RUNTIME_ATTACHED);
     }
 
-    if (SdkVersionPrefixHolder.isRpIntegration()) {
+    // any manual RP integration will support loading user's applcationinsights.json
+    if (SdkVersionPrefixHolder.isRpIntegration()
+        && RpAttachType.getRpAttachType() != RpAttachType.MANUAL) {
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java:484]: would this work?
```suggestion
    if (RpAttachType.getRpAttachType() == RpAttachType.AUTO) {
```

Comment (heyams) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java:484]: latest exporter has 3 types 

MANUAL("Manual"), // Manually writing code to start collecting telemetry, e.g. via azure-monitor-opentelemetry-exporter
  STANDALONE_AUTO("StandaloneAuto"), // RP attach is enabled via a custom JAVA_OPTS
  INTEGRATED_AUTO("IntegratedAuto"); // RP attach is on by default

Comment (heyams) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java:484]: beta.12 will have this..  no config is allowed for non MANUAL

Comment (trask) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java:484]: how about then?
```suggestion
    if (RpAttachType.getRpAttachType() != RpAttachType.MANUAL) {
```

Comment (heyams) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java:484]: getRpAttachType will run into assertError in the non-attach scenario setRpAttachType is never called

Comment (heyams) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java:484]: getRpAttachType will return null in that case if we don't throw assertError

Comment (heyams) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java:484]: we need to check both

Comment (trask) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java:484]: > getRpAttachType will run into assertError in the non-attach scenario setRpAttachType is never called

can we set it in all cases?

**Comment by heyams** [comment]
i've tested it in AKS env, it works.

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java [hunk 2]
```diff
@@ -478,7 +479,8 @@ private static Configuration loadConfigurationFile(Path agentJarPath) {
       return getConfiguration(runtimeAttachedConfigurationContent, JsonOrigin.RUNTIME_ATTACHED);
     }
 
-    if (SdkVersionPrefixHolder.isRpIntegration()) {
+    // any manual RP integration will support loading user's applcationinsights.json
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java:482]: (or do we need this, see comment in line below)
```suggestion
    // only RP integrations do not support loading applicationinsights.json
```

**Review by trask — APPROVED** [review]

**Pull request merged**